learnify/mobile/lib/features/auth/login/view-model/login_view_model.dart

```diff
@@ -0,0 +1,98 @@
+import 'dart:async';
+
+import 'package:async/async.dart';
+import 'package:flutter/material.dart';
+
+import '../../../../core/base/view-model/base_view_model.dart';
+import '../../../../core/managers/network/models/l_response_model.dart';
+import '../../../../product/constants/navigation_constants.dart';
+import '../../service/auth_service.dart';
+import '../../service/l_auth_service.dart';
+import '../model/login_request_model.dart';
+import '../model/login_response_model.dart';
+
+/// View model to manage the data on login screen.
+class LoginViewModel extends BaseViewModel {
+  late final IAuthService _authService;
+
+  late TextEditingController _emailController;
+  late TextEditingController _passwordController;
+
+  TextEditingController get emailController => _emailController;
+  TextEditingController get passwordController => _passwordController;
+
+  late GlobalKey<FormState> _formKey;
+  GlobalKey<FormState> get formKey => _formKey;
+
+  bool _canLogin = false;
+  bool get canLogin => _canLogin;
+
+  @override
+  void initViewModel() {
+    _authService = AuthService.instance;
+  }
+
+  @override
+  void initView() {
+    _emailController = TextEditingController();
+    _passwordController = TextEditingController();
+    _formKey = GlobalKey<FormState>();
+    _emailController.addListener(_controllerListener);
+    _passwordController.addListener(_controllerListener);
+    _setDefault();
+  }
+
+  @override
+  void disposeView() {
+    _emailController.dispose();
+    _passwordController.dispose();
+    _setDefault();
+    super.disposeView();
+  }
+
+  void _controllerListener() {
+    final bool newCanLogin =
+        _passwordController.text.isNotEmpty && _emailController.text.isNotEmpty;
+    if (_canLogin == newCanLogin) return;
+    _canLogin = newCanLogin;
+    notifyListeners();
+  }
+
+  /// login callback.
+  Future<String?> login() async {
+    await operation?.cancel();
+    operation = CancelableOperation<String?>.fromFuture(_loginRequest());
+    final String? res = await operation?.valueOrCancellation();
+    return res;
+  }
+
+  Future<String?> _loginRequest() async {
+    final bool isValid = formKey.currentState?.validate() ?? false;
+    if (isValid) {
+      final LoginRequest requestModel = LoginRequest(
+          email: _emailController.text, password: _passwordController.text);
+      final IResponseModel<LoginResponse> res =
+          await _authService.login(requestModel);
+      if (res.hasError) return res.error?.errorMessage;
+      await navigationManager.navigateToPageClear(
+          path: NavigationConstants.home);
```

Comment:
You shouldn't navigate to home when unverified users try to login. Instead, redirect them to verify screen. You can do that here once you get the response you can add a check for `if(res.data?.user?.isVerified == false)`

Reply:
That was an important case. I have changed accordingly. Thank you, @ezgy !